lib/ClientConnection.cc

```diff
@@ -1596,7 +1596,11 @@ void ClientConnection::close(Result result) {
     }
 
     lock.unlock();
-    LOG_INFO(cnxString_ << "Connection closed with " << result);
+    if (result != ResultDisconnected) {
+        LOG_ERROR(cnxString_ << "Connection closed with " << result);
```

Comment:
When the result is `ResultRetryable`, is there no need to print the error log here?

BTW: I think just print the info level log here because calling close is the expected behavior of the code.

Reply:
Thanks. I changed it to print info log when the result is ResultRetryable.
> calling close is the expected behavior of the code.
Many calling of the close is unexpected behavior. such as：

https://github.com/apache/pulsar-client-cpp/blob/5d5ed2acc2e89aed90b593343369f360644440e7/lib/ClientConnection.cc#L485-L491

https://github.com/apache/pulsar-client-cpp/blob/5d5ed2acc2e89aed90b593343369f360644440e7/lib/ClientConnection.cc#L213-L219

https://github.com/apache/pulsar-client-cpp/blob/5d5ed2acc2e89aed90b593343369f360644440e7/lib/ClientConnection.cc#L223-L229

etc...

Reply:
As you provided the example, The call `close` may have been an unexpected behavior, so we printed the error log before the call. But inside the `close` method, the execution close method behaves as expected and is closed correctly.

Reply:
It will pass the default result value `ResultConnectError` to the `close()` to indicate that the connection is closed unexpectedly. 
In addition, the info log with the `ConnectError` message seems confusing.

Reply:
Okay, I agree.